src/Compilers/VisualBasic/Portable/Binding/InitializerSemanticModel.vb

```diff
@@ -97,12 +97,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                         Debug.Assert(initializer IsNot Nothing)
                         If initializer.Kind = SyntaxKind.EqualsValue Then
                             Dim enumSymbol = DirectCast(Me.MemberSymbol, SourceEnumConstantSymbol)
-                            boundInitializer = binder.BindFieldAndEnumConstantInitializer(enumSymbol, DirectCast(initializer, EqualsValueSyntax), isEnum:=True, diagnostics:=diagnostics, constValue:=Nothing)
+                            boundInitializer = WrapWithEqualsValue(binder.BindFieldAndEnumConstantInitializer(enumSymbol, DirectCast(initializer, EqualsValueSyntax), isEnum:=True, diagnostics:=diagnostics, constValue:=Nothing), initializer)
```

Comment:
This wrapping occurs in a couple of specific places rather than in all cases because some cases produce initializers that are not expressions.